### PR TITLE
Toggle batch normalisation

### PR DESCRIPTION
**Pull request by alecokas**
Nothing much, just adding the ability to turn off batch norm

### CODE REVIEW

**Comment by alecokas** [comment]
Solves issue #19

File: models/discriminators.py [hunk 1]
```diff
@@ -76,14 +76,14 @@ def __init__(self, img_size, kernel_size, num_filters, lr, w_init, bn_init):
...
         x = self.leaky_relu_1(x)
 
-        x = self.conv_block_1(x)
-        x = self.conv_block_2(x)
-        x = self.conv_block_3(x)
+        x = self.conv_block_1(x, training=training)
+        x = self.conv_block_2(x, training=training)
+        x = self.conv_block_3(x, training=training)
 
         res = self.res_block(x)
```
Comment (Devin-Taylor) [inline models/discriminators.py:88]: Should be passed into here too, res blocks contain batchnorm

File: models/generators.py [hunk 2]
```diff
@@ -158,7 +158,10 @@ def loss(self, predictions_on_fake, mean, log_sigma):
             predictions for the images generated by this model.
         """
         kl_coeff = 1
-        loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(labels=tf.ones_like(predictions_on_fake), logits=predictions_on_fake))
+        loss = tf.reduce_mean(
+            tf.nn.sigmoid_cross_entropy_with_logits(
+                labels=tf.ones_like(predictions_on_fake), logits=predictions_on_fake
+            ))
```
Comment (Devin-Taylor) [inline models/generators.py:164]: this fight will never end haha

**Comment by alecokas** [comment]
Added that update

File: models/generators.py [hunk 3]
```diff
@@ -125,12 +125,12 @@ def __init__(self, img_size, kernel_size, num_filters,
...
+    def call(self, embedding, noise, training=True):
         smoothed_embedding, mean, log_sigma = self.conditional_augmentation(embedding)
         noisy_embedding = tf.concat([noise, smoothed_embedding], 1)
 
         x = self.dense_1(noisy_embedding)
-        x = self.bn_1(x)
+        x = self.bn_1(x, training=training)
         x = self.reshape_layer(x)
 
         res_1 = self.res_block_1(x)
```
Comment (Devin-Taylor) [inline models/generators.py:136]: you need to make the change in the generator too mate :joy:
